pkg/services/repo_service.go

```diff
@@ -87,12 +86,41 @@ func (a *argoCDService) GetPaths(ctx context.Context, repoURL string, revision s
 		return nil, err
 	}
 
-	paths, err := gitRepoClient.LsFiles(pattern)
-	if err != nil {
-		return nil, errors.Wrap(err, "Error during listing files of local repo")
+	filteredPaths := []string{}
+
+	repoRoot := gitRepoClient.Root()
+
+	if err := filepath.Walk(repoRoot, func(path string, info os.FileInfo, fnErr error) error {
```

Comment:
Have you considered calling `LsFiles` and then use [filepath.Base()|https://golang.org/pkg/path/filepath/#Base] to only keep directores?   We won't have to explictly skip "." and .git.

Reply:
@wtam2018  - Good thinking, and actually I tried this approach first: the problem is that `LsFiles` doesn't list directories, only files, and there doesn't seem to be a switch to enable it show directories (see `man git-ls-files`). `LsFiles` is just a minimal wrapper around the `git ls-files --full-name -- (path to repo)` command, which you can try yourself on a local repo to see that it only lists files.